src/gfx.cpp

```diff
@@ -1185,6 +1190,44 @@ static void GfxBlitter(const Sprite * const sprite, int x, int y, BlitterMode mo
 	BlitterFactory::GetCurrentBlitter()->Draw(&bp, mode, zoom);
 }
 
+/**
+ * Draws a sprite to a new RGBA buffer (see Colour union) instead of drawing to the screen.
+ * Free the memory after use.
+ *
+ * @param spriteId The sprite to draw.
+ * @return Pixel Buffer
+ */
+const uint32* DrawSpriteToRgbaBuffer(SpriteID spriteId)
+{
+	// Gather information about the sprite to write.
+	const SpriteID real_sprite = GB(spriteId, 0, SPRITE_WIDTH);
+	const Sprite *sprite = GetSprite(real_sprite, ST_NORMAL);
+
+	// Prepare new DrawPixelInfo - Normally this would be the screen but we want to draw to another buffer here.
+	// Normally, pitch would be scaled screen width, but in our case our "screen" is only the sprite width wide.
+	DrawPixelInfo dpi;
+	dpi.dst_ptr = MallocT<uint32>(sprite->width * sprite->height);
```

Comment:
This will break when using any 8bpp blitter (like e.g. `8bpp-optimized`), as they output byte-sized palette indices. By default, MacOS will always use one of the 32bpp blitters, but it is still possible for a user to override this.

As such, this situation should be gracefully handled, especially as later on somebody else might trust the function name and use it for something else not specific to MacOS, not realising this limitation.